### PR TITLE
Implement ability to individually enable/disable Enchantments via con…

### LINKED ISSUE #1580: [1.16.4] [Suggestion] Change enchant config to allow each one to be disabled individually
Minecraft Version: 1.16.4

I am submitting this as a suggestion.
I would like to be able to disable certain enchants such as magnet or growth without disabling all of the enchantments.

### PR DESCRIPTION
**Pull request by trunksbomb**
…figuration. Closes #1580

### CODE REVIEW

**Comment by trunksbomb** [comment]
We both implemented the same thing, more or less the same way. Oops.

Although I accidentally snuck a modified build.gradle in there.

**Comment by Lothrazar** [comment]
oh sorry, i didnt know you were working on this. 

Normally i try to "assign myself" on a bug to show im doing it, i forgot to this time, my bad

File: build.gradle [hunk 1]
```diff
@@ -106,6 +105,8 @@ minecraft {
 //}
 
 dependencies {
+    implementation 'org.jetbrains:annotations:19.0.0'
```
Comment (Lothrazar) [inline build.gradle:108]: What does this provide? I can add to main branch if its useful

File: src/main/java/com/lothrazar/cyclic/registry/EnchantRegistry.java [hunk 2]
```diff
@@ -32,23 +35,37 @@
...
+    registerIfEnabled(r, new EnchantAutoSmelt(Enchantment.Rarity.RARE, EnchantmentType.DIGGER, EquipmentSlotType.MAINHAND).setRegistryName("auto_smelt"));
+    registerIfEnabled(r, new EnchantDisarm(Enchantment.Rarity.UNCOMMON, EnchantmentType.WEAPON, EquipmentSlotType.MAINHAND).setRegistryName("disarm"));
+    registerIfEnabled(r, new EnchantCurse(Enchantment.Rarity.RARE, EnchantmentType.ARMOR, EquipmentSlotType.CHEST).setRegistryName("curse"));
+    registerIfEnabled(r, new EnchantPearl(Enchantment.Rarity.RARE, EnchantmentType.WEAPON, EquipmentSlotType.MAINHAND).setRegistryName("ender"));
+    registerIfEnabled(r, new EnchantBeekeeper(Enchantment.Rarity.VERY_RARE, EnchantmentType.ARMOR_HEAD, EquipmentSlotType.HEAD).setRegistryName("beekeeper"));
+  }
+
+  private static void registerIfEnabled(IForgeRegistry<Enchantment> r, Enchantment enchantment) {
+    try {
+      if (((ForgeConfigSpec.BooleanValue)enchantment.getClass().getDeclaredField("ENABLED").get(enchantment)).get()) {
```
Comment (Lothrazar) [inline src/main/java/com/lothrazar/cyclic/registry/EnchantRegistry.java:60]: Looks like we had the exact same idea. only difference is i used a base class , but this does the job too

File: build.gradle (hunk 1 above)
Comment (trunksbomb) [reply build.gradle:108]: TBH I'm not sure how that got there.

**Comment by trunksbomb** [comment]
Closing PR because we both implemented this but yours is already committed.

**Pull request closed without merging**